spec/rspec/core/dsl_spec.rb

```diff
@@ -6,75 +6,30 @@
   include RSpec::Support::InSubProcess
 
   shared_examples_for "dsl methods" do |*method_names|
-    context "when expose_dsl_globally is enabled" do
-      def enable
-        in_sub_process do
-          changing_expose_dsl_globally do
-            RSpec.configuration.expose_dsl_globally = true
-            expect(RSpec.configuration.expose_dsl_globally?).to eq true
-          end
-
-          yield
-        end
-      end
-
-      it 'makes them only available off of `RSpec`, `main` and modules' do
-        enable do
-          expect(::RSpec).to respond_to(*method_names)
-          expect(main).to respond_to(*method_names)
-          expect(Module.new).to respond_to(*method_names)
-
-          expect(Object.new).not_to respond_to(*method_names)
-        end
-      end
-    end
-
-    context "when expose_dsl_globally is disabled" do
-      def disable
-        in_sub_process do
-          changing_expose_dsl_globally do
-            RSpec.configuration.expose_dsl_globally = false
-            expect(RSpec.configuration.expose_dsl_globally?).to eq false
-          end
-
-          yield
-        end
-      end
+    it 'are only available off of `RSpec`' do
+      in_sub_process do
+        setup
 
-      it 'makes them only available off of `RSpec`' do
-        disable do
-          expect(::RSpec).to respond_to(*method_names)
+        expect(::RSpec).to respond_to(*method_names)
 
-          expect(main).not_to respond_to(*method_names)
-          expect(Module.new).not_to respond_to(*method_names)
-          expect(Object.new).not_to respond_to(*method_names)
-        end
+        expect(main).not_to respond_to(*method_names)
+        expect(Module.new).not_to respond_to(*method_names)
+        expect(Object.new).not_to respond_to(*method_names)
       end
     end
   end
 
   describe "built in DSL methods" do
     include_examples "dsl methods", :describe, :context, :shared_examples, :shared_examples_for, :shared_context do
-      def changing_expose_dsl_globally
-        yield
+      def setup
       end
     end
   end
 
   describe "custom example group aliases" do
     context "when adding aliases before exposing the DSL globally" do
       include_examples "dsl methods", :detail do
-        def changing_expose_dsl_globally
-          RSpec.configuration.alias_example_group_to(:detail)
-          yield
-        end
-      end
-    end
-
-    context "when adding aliases after exposing the DSL globally" do
-      include_examples "dsl methods", :detail do
-        def changing_expose_dsl_globally
-          yield
+        def setup
           RSpec.configuration.alias_example_group_to(:detail)
         end
```

Comment:
Can you not just do this in a before hook here? Then you don't need the empty method call?

Reply:
I've started off doing exactly as you say, but it turns out `before` would run outside of `in_sub_process`, and thus litter the configuration.

It would be possible to go with `before` if `in_sub_process` could be done in `around { |example| in_sub_process { example.run } }`. But it didn't work for me just recently in [this PR](https://github.com/rspec/rspec-support/pull/452), all examples were marked as pending.
I decided to postpone the investigation on why this is the case, and use a safe approach for now.